collections/_posts/2023-04-11-SAM.md

```diff
@@ -0,0 +1,156 @@
+
+## Edge detection
+
+A 16x16 grid of points is applied to all images which leads to 768 masks per image (3 per point). The redundancy is removed with Non-Maximal Suppression (NMS) and a Sobel filter is applied on un-thresholded masks probability masks to have edges. 
+
+![](/collections/images/SAM/edge_detection.jpg)
+
+![](/collections/images/SAM/edge_detection_table.jpg)
+
+Even if SAM is not trained for that the results are not bad but includes sensible edges that were not even annotated.
+
+## Object proposal
+
+The mask generation pipeline is modified to make mask proposals. They concentrate on LVIS dataset as there are many categories and compare SAM to a ViTDet detector.
+
+![](/collections/images/SAM/object_proposal.jpg)
+
+SAM is not as good but is competitive on big objects.
+
+## Instance segmentation
+
+The boxes of ViTDet are used as prompt on COCO and LVIS datasets. It is an example of compostion of SAM to a larger system.
+
+![](/collections/images/SAM/instance_segmentation.jpg)
+
+![](/collections/images/SAM/mask_quality_rating_COCO_LVIS.jpg)
+
+In comparison with ViTDet-H, SAM's performance is not as good but is not far. According to human rates it is better. The case of COCO is particular as the ground truth has many defaults, that are learned by the network, what SAM doesn't do.
+
+## Text to mask
+
+Segmenting an image from text input is a proof of concept of SAM’s ability to process text prompt. This time SAM is trained differently but without the need of new text annotation. For each masks with an area $$>100^2$$, the CLIP image embedding is extracted and during training those embeddings are prompted at its first interaction. During the inference the text is run through CLIP’s text encoder which is then prompted to SAM.
+
+![](/collections/images/SAM/text_to_mask.jpg)
+
+## Ablations
+
+The ablations concern the SA-1B dataset and the architecture of SAM.
+
+![](/collections/images/SAM/ablation_study.jpg)
+
+The performance increase through all the data engine steps but in the end only images with automatically generated masks are kept to simplify setup training.
+
+# Discussion
+
+Models are almost always dedicated to a particular task, but here the foundation model trained on a very large dataset is then adapted to many tasks. However, generally foundation models are auto-supervised wherease here it is initialised with MAE that is based on a supervised work.
```

Comment:
wherease -> whereas
+
To clear the confusion with MAE being supervised or not as pointed out by Gaspard, I would suggest to rephrase things in the folloing way:
_it is initialised with a MAE trained using supervised labels._